big_tests/tests/inbox_SUITE.erl

```diff
@@ -761,9 +802,270 @@ groupchat_markers_all_reset_room_created(Config) ->
+      total = 1,
+      convs = [#conv{unread = 0, from = KateJid, to = BobRoomJid, content = Msg}]},
+                #{case_sensitive => true, check_resource => false})
+    end).
+
+
+
 %%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%
 %% Helpers
 
+go_offline(User, Room, Occupants) ->
+    UnavailavbleStanza = escalus_stanza:presence(<<"unavailable">>),
+    Stanza = muc_helper:stanza_to_room(UnavailavbleStanza, Room, nick(User)),
+    escalus:send(User, Stanza),
+    lists:foreach(fun(User) -> A = escalus:wait_for_stanza(User), ct:pal("A: ~p", [A]) end,
+                  Occupants).
+
+go_online(User, Room, Occupants) ->
+    UnavailavbleStanza = escalus_stanza:presence(<<"unavailable">>),
+    Stanza = muc_helper:stanza_to_room(UnavailavbleStanza, Room, nick(User)),
+    escalus:send(User, Stanza),
+    lists:foreach(fun(User) -> A = escalus:wait_for_stanza(User), ct:pal("A: ~p", [A]) end,
+                  Occupants).
+
+wait_for_groupchat_msg(Users) ->
+    Resps = lists:map(fun(User) -> escalus:wait_for_stanza(User) end,
+              Users),
+    lists:foreach(fun(Resp) -> escalus:assert(is_groupchat_message, Resp) end,
+                  Resps).
+
+
+to_bare_lower(User) ->
+    lbin(escalus_client:short_jid(User)).
+
+make_members(Room, Admin, Users) ->
+    Items = lists:map(fun(User) -> {escalus_utils:get_short_jid(User),<<"member">>} end,
+                      Users),
+    escalus:send(Admin, stanza_set_affiliations(Room, Items)),
+    SuccesResp = escalus:wait_for_stanzas(Admin, 1 + length(Users)), % gets iq result and affs changes from all users
+    ct:pal("SuccessResp: ~p", [SuccesResp]),
+    lists:foreach(fun(User) -> escalus:wait_for_stanzas(User, length(Users)) end, Users). % Everybody gets aff changes of everybody
+
+% All users enter the room
+enter_room(Room, Users) ->
+    lists:foreach(fun(User) ->
+                          escalus:send(User, stanza_muc_enter_room(Room, nick(User))) end,
+                  Users),
+    lists:foreach(fun(User) ->
+                          A = escalus:wait_for_stanzas(User, length(Users) + 1), % everybody gets presence from everybody + a subject message
+                          ct:pal("For ~p: ~p", [escalus_client:short_jid(User), A])
+                  end,
+                  Users).
+
+% `User` enters the room `Room` where `Users` are occupants
+enter_room(Room, User, Users, DelayedMessagesCount) ->
+    escalus:send(User, stanza_muc_enter_room(Room, nick(User))),
+    lists:foreach(fun(User) ->
+                          A = escalus:wait_for_stanza(User), ct:pal("~p: Got presence from user: ~p", [escalus_client:short_jid(User), A]) end,
+                  Users),
+    Subj = escalus:wait_for_stanzas(User, length(Users) + 1 + 1 + DelayedMessagesCount), % User gets subject message and presences from everybody including himself
+    ct:pal("Subj and presence messages: ~p", [Subj]).
+
+
+stanza_set_affiliations(Room, List) ->
+    Payload = lists:map(fun({JID, Affiliation}) ->
```

Comment:
Can we separate this fun and name it? 
BTW seems that currently the branch ` ({JID, Affiliation, Reason})` won't be called since the `List` variable will be the result only of :
```
Items = lists:map(fun(User) -> {escalus_utils:get_short_jid(User),<<"member">>} end, Users),
escalus:send(Admin, stanza_set_affiliations(Room, Items)),
```
So it's always tuple of two elements